### PR TITLE
Remove deprecated method getTableHandle from spi

### PR DESCRIPTION
**Pull request by chenjian2664**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by ebyhr — CHANGES_REQUESTED** [review]
Connectors that don't support versioned query should deny the operations. Try:
```sql
SELECT * FROM tpch.tiny.region FOR VERSION AS OF 1;
```

**Comment by ebyhr** [comment]
Please confirm CI failure.

File: core/trino-main/src/main/java/io/trino/connector/system/SystemTablesMetadata.java [hunk 1]
```diff
@@ -63,12 +65,17 @@ public List<String> listSchemaNames(ConnectorSession session)
...
     @Override
-    public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
+    public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName, Optional<ConnectorTableVersion> startVersion, Optional<ConnectorTableVersion> endVersion)
     {
         Optional<SystemTable> table = tables.getSystemTable(session, tableName);
         if (table.isEmpty()) {
             return null;
         }
+
+        if (startVersion.isPresent() || endVersion.isPresent()) {
```
[the remark this answers is not in the document]
Comment (chenjian2664) [reply core/trino-main/src/main/java/io/trino/connector/system/SystemTablesMetadata.java:75]: There are tests fail with the the check, such as: `testTrySelectTableVersion`, `testCreateOrReplaceTableSnapshots`, `testReadingFromSpecificSnapshot` in Iceberg.

Add behind is compatible with preview logic, but I am not strongly about it.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 2]
```diff
@@ -4523,4 +4523,17 @@ protected void verifyVersionedQueryFailurePermissible(Exception e)
                         "Delta Lake snapshot ID does not exists: .*|" +
                         "Unsupported type for table version: .*");
     }
+
+    @Test
+    @Override
+    public void testSelectVersionOfNonExistentTable()
```
[the remark this answers is not in the document]
Comment (chenjian2664) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:4529]: How to distinguish the connector support version tables or not, shall we add a testing behavior?

File: core/trino-main/src/main/java/io/trino/connector/system/SystemTablesMetadata.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chenjian2664) [reply core/trino-main/src/main/java/io/trino/connector/system/SystemTablesMetadata.java]: @ebyhr Move this line in front of the calling,  there are all the test failures in Iceberg and Deltalake

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (chenjian2664) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:4529]: Add a testing behavior

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java [hunk 3]
```diff
@@ -137,14 +120,6 @@ default ConnectorTableHandle getTableHandle(
...
     {
-        ConnectorTableHandle tableHandle = getTableHandle(session, tableName);
-        if (tableHandle == null) {
-            // Not found
-            return null;
-        }
-        if (startVersion.isEmpty() && endVersion.isEmpty()) {
-            return tableHandle;
-        }
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support versioned tables");
```
[the remark this answers is not in the document]
Comment (chenjian2664) [reply core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java:123]: updated

**Comment by chenjian2664** [comment]
https://github.com/trinodb/trino/actions/runs/9028245239/job/24808401292?pr=21532 cc @wendigo

**Comment by chenjian2664** [comment]
@ebyhr PTAL

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 4]
```diff
@@ -690,9 +691,16 @@ public void testSelectInTransaction()
     @Test
     public void testSelectVersionOfNonExistentTable()
     {
+        String tableName = "foo_" + randomNameSuffix();
+
+        if (!hasBehavior(SUPPORTS_VERSIONED_TABLE)) {
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:696]: We could change the assertion to `hasMessageMatching` instead. 

Introducing `SUPPORTS_VERSIONED_TABLE` requires more work. For instance, javadoc in testTrySelectTableVersion needs update.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**